driver/ppm_fillers.c

```diff
@@ -2641,6 +2644,64 @@ static int f_sys_poll_e(struct event_filler_arguments *args)
 	return add_sentinel(args);
 }
 
+static int timespec_parse(struct event_filler_arguments *args, unsigned long val)
+{
+	uint64_t longtime;
+	char *targetbuf = args->str_storage;
+	struct timespec *tts = (struct timespec *)targetbuf;
+	int cfulen;
+
+	/*
+	 * interval
+	 * We copy the timespec structure and then convert it to a 64bit relative time
+	 */
+	cfulen = (int)ppm_copy_from_user(targetbuf, (void __user *)val, sizeof(struct timespec));
+
+	if (unlikely(cfulen != 0))
+		return PPM_FAILURE_INVALID_USER_MEMORY;
+
+	longtime = ((uint64_t)tts->tv_sec) * 1000000000 + tts->tv_nsec;
+
+	return val_to_ring(args, longtime, 0, false, 0);
+}
+
+static int f_sys_ppoll_e(struct event_filler_arguments *args)
+{
+	unsigned long val;
+	int res;
+
+	res = poll_parse_fds(args, true);
+	if (unlikely(res != PPM_SUCCESS))
```

Comment:
likely/unlikely can likely be removed, gregkh suggested that once when reviewing our code, so we're trying to not use them anymore for new code